joblib/test/test_pickle.py

```diff
@@ -0,0 +1,21 @@
+"""
+Test that pickle and dill output the same bytes
+"""
+import pickle
+
+import nose
+
+from joblib.test.common import dill, with_dill, with_py3
+
+
+def global_func():
+    pass
+
+
+@with_py3
+@with_dill
+def test_dumps_eq():
+    # Test that dill and pickle produce the same results
```

Comment:
That's a fair point if we decide to ship dill inside joblib.

The concern is that you don't want to have to recompute everything if you update your joblib version. In some of our use cases, people have joblib caches that would take ~ days to recreate from scratch. 

In this respect we probably care mostly about hashes being consistent across joblib versions.